geomesa-features/geomesa-feature-avro-complex/src/main/scala/org/locationtech/geomesa/features/avro/complex/AvroSchemas.scala

```diff
@@ -0,0 +1,253 @@
+package org.locationtech.geomesa.features.avro.complex
+
+import org.apache.avro.Schema.Field
+import org.apache.avro.Schema.Type._
+import org.apache.avro.io.{Decoder, Encoder}
+import org.apache.avro.{Schema, SchemaBuilder}
+import org.geotools.feature.NameImpl
+import org.geotools.gml3.GMLSchema
+import org.geotools.xs.XSSchema
+import org.locationtech.geomesa.utils.geotools.RichAttributeDescriptors._
+import org.opengis.feature.`type`._
+
+import scala.annotation.tailrec
+import scala.collection.JavaConversions._
+import scala.collection.immutable.{BitSet, Queue}
+import scala.collection.mutable
+
+object AvroSchemas {
+  val FEATURE_ID_AVRO_FIELD_NAME = "__fid__"
+  val AVRO_SIMPLE_FEATURE_VERSION = "__version__"
+  val AVRO_SIMPLE_FEATURE_USERDATA = "__userdata__"
+  val VERSION = 4

```

Comment:
Ok, just for my own education and clarity.  Who's version is version 4 here?

Avro?  GeoMesa's Avro schema?  OGC's ComplexFeature?